src/libgpu/src/vulkan/vulkan_surface.cpp

```diff
@@ -181,24 +181,24 @@ Driver::_copySurface(SurfaceObject *dst, SurfaceObject *src, SurfaceSubRange ran
 
    vk::ImageCopy copyRegion;
    if (src->desc.dim == latte::SQ_TEX_DIM::DIM_3D) {
-      copyRegion.srcOffset = { 0, 0, static_cast<int32_t>(range.firstSlice) };
+      copyRegion.srcOffset = VkOffset3D { 0, 0, static_cast<int32_t>(range.firstSlice) };
```

Comment:
Interesting.  This shouldn't be needed I don't think?  Why does it fail?

Reply:
```
/home/user/decaf-emu/src/libgpu/src/vulkan/vulkan_surface.cpp: In member function ‘vulkan::SurfaceObject* vulkan::Driver::_allocateSurface(const vulkan::SurfaceDesc&)’:
/home/user/decaf-emu/src/libgpu/src/vulkan/vulkan_surface.cpp:480:41: error: ambiguous overload for ‘operator=’ (operand types are ‘vk::Offset3D’ and ‘<brace-enclosed initializer list>’)
    bufferRegion.imageOffset = { 0, 0, 0 };
                                         ^
In file included from /home/user/decaf-emu/src/libgpu/./gpu_vulkandriver.h:4,
                 from /home/user/decaf-emu/src/libgpu/src/vulkan/vulkan_driver.h:5,
                 from /home/user/decaf-emu/src/libgpu/src/vulkan/vulkan_surface.cpp:2:
/usr/include/vulkan/vulkan.hpp:4948:15: note: candidate: ‘vk::Offset3D& vk::Offset3D::operator=(const VkOffset3D&)’
     Offset3D& operator=( VkOffset3D const & rhs )
               ^~~~~~~~
/usr/include/vulkan/vulkan.hpp:4925:10: note: candidate: ‘constexpr vk::Offset3D& vk::Offset3D::operator=(const vk::Offset3D&)’
   struct Offset3D
          ^~~~~~~~
/usr/include/vulkan/vulkan.hpp:4925:10: note: candidate: ‘constexpr vk::Offset3D& vk::Offset3D::operator=(vk::Offset3D&&)’

```

Am I using to old libvulkan?

Reply:
I'm using the latest SDK (effectively, I custom built it for transform feedback).  That's a stupid warning though, since the ambiguous overload it mentions are for the same thing....  They both initialize the same vk type >_<.

Reply:
In hindsight, I don't think the extra specificity is actually a bad thing.  You will note that in almost every case, I try to avoid using initializer lists anyways, since the parameters being set are so unclear when you read.

Reply:
I checked my copy in of Offset3D constructors and the one in master branch of Vulkan-Hpp, and they look identical:

    Offset3D( int32_t x_ = 0,
              int32_t y_ = 0,
              int32_t z_ = 0 );
    explicit Offset3D( Offset2D const& offset2D,
                       int32_t z_ = 0 );
    Offset3D( VkOffset3D const & rhs );

and operator = is also the same:

    Offset3D& operator=( VkOffset3D const & rhs );


so I think it is not related to version of vulkan.

The problem is that the compiler do not know which version to call operator=( VkOffset3D const & rhs ); or compiler provided operator= methods.

This is because due to implicit argument type conversion. the brace-enclosed init list can both be passed to constructors of VkOffset3D and vk::Offset3D.  (this is because VkOffset3D has trivial constructor that can do it).

Yes, the bodies and semantic are the exactly same, but compiler doesn't know that. So you just need to select one you prefer.

I can make it vk::Offset3D { ... }.  I guess it might be more consistent with rest of the code.

I just tested it and it also work.

Reply:
Yup.  Use vk::Offset3D and it will be good.  Then just the commit squash / message and I can merge once CI finishes :)